src/components/Tabs/tests/Tabs.test.tsx

```diff
@@ -193,4 +206,245 @@ describe('<Tabs />', () => {
       expect(spy).toHaveBeenCalledWith(1);
     });
   });
+
+  describe('<TabMeasurer />', () => {
+    const mockTabs = [
+      {content: 'Tab 1', id: 'tab-1'},
+      {content: 'Tab 2', id: 'tab-2'},
+      {content: 'Tab 3', id: 'tab-3'},
+    ];
+
+    it('is not set to anything by default', () => {
+      const tabs = mountWithAppProvider(<Tabs {...mockProps} />);
+      expect(tabs.find(TabMeasurer).prop('tabToFocus')).toBe(-1);
+    });
+
+    it('resets focus when selected gets updated to the previously selected value', () => {
+      const tabs = mountWithAppProvider(
+        <Tabs {...mockProps} selected={0} tabs={mockTabs} />,
+      );
+      trigger(tabs.find('ul'), 'onKeyUp', {
+        key: 'ArrowRight',
+      });
+      tabs.setProps({selected: 0});
+      expect(tabs.find(TabMeasurer).prop('tabToFocus')).toBe(-1);
+    });
+
+    it('passes the correct selected value if given', () => {
+      const tabs = mountWithAppProvider(
+        <Tabs {...mockProps} selected={1} tabs={mockTabs} />,
+      );
+      expect(tabs.find(TabMeasurer).prop('selected')).toBe(1);
+    });
+
+    describe('ArrowRight', () => {
+      it('shifts focus to the next tab when pressing ArrowRight', () => {
+        const tabs = mountWithAppProvider(
+          <Tabs {...mockProps} tabs={mockTabs} />,
+        );
+        trigger(tabs.find('ul'), 'onKeyUp', {
+          key: 'ArrowRight',
+        });
+        expect(tabs.find(TabMeasurer).prop('tabToFocus')).toBe(0);
+      });
+
+      it('shifts focus to the first tab when pressing ArrowRight on the last tab', () => {
+        const tabs = mountWithAppProvider(
+          <Tabs {...mockProps} tabs={mockTabs} />,
+        );
+        let x;
+        for (x = 0; x <= 3; x++) {
```

Comment:
I think maybe we should just repeat the action three times. Loops always feel a bit black box-ish in test imo.